File: src/main/java/net/theevilreaper/vulpes/api/repository/SoundRepository.java
package net.theevilreaper.vulpes.api.repository;

import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.PageableRepository;
import net.theevilreaper.vulpes.api.model.sound.SoundModel;

import java.util.UUID;

@Repository
public interface SoundRepository extends PageableRepository<SoundModel, UUID> {
}

File: src/main/kotlin/net/theevilreaper/vulpes/api/repository/SoundRepository.kt
package net.theevilreaper.vulpes.api.repository

import net.theevilreaper.vulpes.api.model.sound.SoundModel
import org.springframework.data.mongodb.repository.MongoRepository

/**
 * The repository is used to store different object instances from the [SoundModel].
 * It extends the [MongoRepository] to provide the basic CRUD operations.
 * @see MongoRepository
 * @author theEvilReaper
 */
interface SoundRepository : MongoRepository<SoundModel, String>